### ISSUE TITLE
full day events not rendering in ios in month view & in week & day view onPressEvent is not working

### ISSUE THREAD

**Issue by girish-floorzap** [acro5piano/react-native-big-calendar#575]
Hi,

I am facing issue with full day event rendering in month view in ios. And also it is getting rendered in week & day view, but I am not getting the OnPressEvent on click of that event. Please find the attached images

![Screenshot 2021-12-23 at 2 32 51 PM](https://user-images.githubusercontent.com/96039705/147216046-39741a32-deb9-41ce-8158-529f72d6732a.png)
![Screenshot 2021-12-23 at 2 32 31 PM](https://user-images.githubusercontent.com/96039705/147216057-06859420-2cad-4a92-b204-4621eb731823.png)
![Screenshot 2021-12-23 at 2 32 15 PM](https://user-images.githubusercontent.com/96039705/147216066-ded01714-9523-483c-81e6-9833a4e9e6ec.png)
![Screenshot 2021-12-23 at 2 30 41 PM](https://user-images.githubusercontent.com/96039705/147216072-e17d5119-0a1c-45fe-bd52-e61eaaa26169.png)
.

**Comment by acro5piano** [comment]
Thanks for filing this.

1) Can you share your code?

2) The problem is the following, right?

- All day events are not rendered on the month view
- `onPressEvent` is not fired for all day events

for latter, we should modify the code:

https://github.com/acro5piano/react-native-big-calendar/blob/8cdd4226264b4d590b1457bfdbd61d37b036303f/src/components/CalendarHeader.tsx#L143-L155

**Comment by girish-floorzap** [comment]
Yes, below 2 are the problems.
1. All day events are not rendered on the month view
2. onPressEvent is not fired for all day events


Below is my code
 <Calendar
                    height={Dimensions.get('window').height - 60}
                    events={[...calendarEvents]}
                    mode={mode}
                    swipeEnabled={true}
                    eventCellStyle={() => getEventColor()}
                    onPressEvent={(event) => handleEventClick(event)}
                />

below is my event json
'[{"title":"RX-1001","start":"2021-12-14T18:30:00.000Z","end":"2021-12-14T18:30:00.000Z","invoiceId":3,"serviceTypeId":2},{"title":"RX-1002","start":"2021-12-27T18:30:00.000Z","end":"2021-12-30T18:30:00.000Z","invoiceId":4,"serviceTypeId":5},{"title":"RX-1003","start":"2021-12-29T18:30:00.000Z","end":"2021-12-30T18:30:00.000Z","invoiceId":5,"serviceTypeId":8},{"title":"RX-1003","start":"2021-12-21T18:30:00.000Z","end":"2021-12-22T18:30:00.000Z","invoiceId":5,"serviceTypeId":9},{"title":"RX-1004","start":"2022-01-01T18:30:00.000Z","end":"2022-01-04T18:30:00.000Z","invoiceId":6,"serviceTypeId":15},{"title":"RX-1004","start":"2021-12-23T18:30:00.000Z","end":"2021-12-23T18:30:00.000Z","invoiceId":6,"serviceTypeId":16},{"title":"RX-1005","start":"2021-12-25T18:30:00.000Z","end":"2021-12-25T18:30:00.000Z","invoiceId":8,"serviceTypeId":23},{"title":"PO-1001","start":"2021-12-28T18:30:00.000Z","end":"2021-12-28T18:30:00.000Z","invoiceId":9,"serviceTypeId":25},{"title":"RX-1006","start":"2021-12-30T18:30:00.000Z","end":"2021-12-30T18:30:00.000Z","invoiceId":10,"serviceTypeId":27},{"title":"PO-1002","start":"2021-12-20T18:30:00.000Z","end":"2021-12-20T18:30:00.000Z","invoiceId":11,"serviceTypeId":29}]'

Thanks.

**Comment by acro5piano** [comment]
Thanks for sharing the code but I think you are uploading wrong event json. All day events should have time which is 0 in hour, minutes, and seconds (`00:00:00`)

Anyway I can successfully reproduce the issue by following code. I'm debugging now.

```tsx
storiesOf('reproduction-issue-575', module)
  .add('month mode', () => (
    <View style={styles.mobile}>
      <Calendar
        height={MOBILE_HEIGHT}
        events={events}
        mode="month"
        onPressEvent={(event) => alert(event.title)}
      />
    </View>
  ))
  .add('week mode', () => (
    <View style={styles.mobile}>
      <Calendar
        height={MOBILE_HEIGHT}
        events={events}
        mode="week"
        onPressEvent={(event) => alert(event.title)}
      />
    </View>
  ))

const events = [
  {
    title: 'RX-1003',
    start: dayjs().hour(0).minute(0).second(0).toDate(),
    end: dayjs().hour(0).minute(0).second(0).toDate(),
  },
]

```

**Comment by acro5piano** [comment]
@girish-floorzap 
Hi, I've fixed the issue on v2.9.0. Can you check it is working for you? Thanks!

**Comment by girish-floorzap** [comment]
Yes, It is working as expected now. Thanks a lot for your quick fix.
